src/DashboardService.Data/PriorityRepository.cs

```diff
@@ -18,28 +21,83 @@ public PriorityRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbPriority dbDepartment)
+  public async Task<Guid?> CreateAsync(DbPriority dbPriority)
   {
-    throw new NotImplementedException();
+    _provider.Priorities.Add(dbPriority);
+    await _provider.SaveAsync();
+
+    return dbPriority.Id;
+  }
+
+  public async Task<(List<DbPriority> dbPriorities, int totalCount)> GetAllAsync(GetPrioritiesFilter filter, CancellationToken cancellationToken)
+  {
+    IQueryable<DbPriority> dbPriorities = _provider.Priorities.AsQueryable();
+    
+    if (filter.IsAscendingSort.HasValue)
+    {
+      dbPriorities = filter.IsAscendingSort.Value
+        ? dbPriorities.OrderBy(d => d.Name)
+        : dbPriorities.OrderByDescending(d => d.Name);
+    }
+    
+    if (!string.IsNullOrWhiteSpace(filter.NameIncludeSubstring))
+    {
+      dbPriorities = dbPriorities.Where(d => d.Name.Contains(filter.NameIncludeSubstring));
+    }
+    
+    return (
+      await dbPriorities
+        .Skip(filter.SkipCount)
+        .Take(filter.TakeCount)
+        .ToListAsync(cancellationToken),
+      await dbPriorities.CountAsync(cancellationToken));
+  }
+
+  public async Task<DbPriority> GetAsync(Guid id, CancellationToken cancellationToken = default)
+  {
+    return await _provider.Priorities.FindAsync(id, cancellationToken);
   }
```

Comment:
use AsNoTracking() for all methods which won't change the data

Reply:
++